Public/Src/Deployment/NugetPackages.dsc

```diff
@@ -122,6 +122,54 @@ namespace NugetPackages {
         }).deployment
     });
 
+    const xldbnetcorequalifier : BuildXLSdk.DefaultQualifier = {
+        targetFramework: "netcoreapp3.0",
+        configuration: qualifier.configuration,
+        targetRuntime: "win-x64"
+    };
+
+    const xldbnet472qualifier : BuildXLSdk.DefaultQualifier = {
+        targetFramework: "net472",
+        configuration: qualifier.configuration,
+        targetRuntime: "win-x64"
+    };
+
+    const xldblibrary = pack({
+        id: `${packageNamePrefix}.Xldb`,
+        deployment: {
+            contents: [ 
+                Nuget.createAssemblyLayout(importFrom("BuildXL.Tools").Xldb.withQualifier(xldbnetcorequalifier).dll),
+                Nuget.createAssemblyLayout(importFrom("BuildXL.Tools").Xldb.Proto.withQualifier(xldbnetcorequalifier).dll),
+                Nuget.createAssemblyLayout(importFrom("BuildXL.Utilities").withQualifier(xldbnetcorequalifier).dll),
+                Nuget.createAssemblyLayout(importFrom("BuildXL.Utilities").KeyValueStore.withQualifier(xldbnetcorequalifier).dll),
+                Nuget.createAssemblyLayout(importFrom("BuildXL.Utilities").Collections.withQualifier(xldbnetcorequalifier).dll),
+                Nuget.createAssemblyLayout(importFrom("BuildXL.Utilities").Native.withQualifier(xldbnetcorequalifier).dll),
+                Nuget.createAssemblyLayout(importFrom("BuildXL.Utilities").Storage.withQualifier(xldbnetcorequalifier).dll),

```

Comment:
Its a bummer to include these and the primary BuildXL.Utilities dlls in the public surface area of this dll. Is it possible to avoid this?

Reply:
Danny and I tried to create a minimal set of dlls required for this package. Since we use the key-value store accessor, it relies on the native and storage dlls along with several others (the ones here). Removing any of them will render the nuget package unusable (we tried with a separate playground app that imported the new nuget package)

---
In reply to: [316293225](https://github.com/microsoft/BuildXL/pull/775#discussion_r316293225) [](ancestors = 316293225)